### PR TITLE
Add support for loading DCF configuration to remote node

### PR DESCRIPTION
**Pull request by meifonglow**
As part of instrument commissioning process, I want to apply the configuration values stored in a DCF to a remote node, including PDO mapping and communications parameters.  If a value was not specified, then the EDS default value is used instead.

This commit provides this capability.

Potentially addresses #274.

### CODE REVIEW

File: canopen/node/remote.py [hunk 1]
```diff
@@ -147,8 +147,19 @@ def __load_configuration_helper(self, index, subindex, name, value):
...
     def load_configuration(self):
         ''' Load the configuration of the node from the object dictionary.'''
+        # First apply PDO configuration from object dictionary
+        self.pdo.read(from_od=True)
+        self.pdo.save()
+
+        # Now apply all other records in object dictionary
         for obj in self.object_dictionary.values():
-            if isinstance(obj, ODRecord) or isinstance(obj, ODArray):
+            if "PDO" in obj.name:
```
[the remark this answers is not in the document]
Comment (meifonglow) [reply canopen/node/remote.py:156]: Do you mean explicitly ignoring objects with indices within supported PDO ranges, i.e: `0x1400 <= obj.index < 0x1C00` ?

[the remark this answers is not in the document]
Comment (christiansandberg) [reply canopen/node/remote.py:156]: Yes exactly. I'm thinking using names might yield both false positives and false negatives.

[the remark this answers is not in the document]
Comment (meifonglow) [reply canopen/node/remote.py]: Thanks that makes sense.
I've also removed excluding objects with "COB-ID" keyword.  They are being applied OK by `__load_configuration_helper()` and would have been incorrectly ignored in previous commit.

File: canopen/node/remote.py [hunk 2]
```diff
@@ -147,8 +147,16 @@ def __load_configuration_helper(self, index, subindex, name, value):
...
+        # First apply PDO configuration from object dictionary
+        self.pdo.read(from_od=True)
+        self.pdo.save()
+
+        # Now apply all other records in object dictionary
         for obj in self.object_dictionary.values():
-            if isinstance(obj, ODRecord) or isinstance(obj, ODArray):
+            if 0x1400 <= obj.index < 0x1c00:
+                # Ignore PDO related objects
+                logger.debug(f"{obj.index:#06x}: {obj.name} already applied by pdo.save()")
```
Comment (acolomb) [inline canopen/node/remote.py:158]:
```suggestion
                logger.debug("0x%04X: %s already applied by pdo.save()", obj.index, obj.name)
```

See #443.  Do we really need this much debug logging though?

File: canopen/pdo/base.py [hunk 3]
```diff
@@ -314,11 +314,20 @@ def add_callback(self, callback: Callable[["Map"], None]) -> None:
         self.callbacks.append(callback)
 
     def read(self, from_od=False) -> None:
-        """Read PDO configuration for this map using SDO."""
+        """Read PDO configuration for this map
```
Comment (acolomb) [inline canopen/pdo/base.py:317]:
```suggestion
        """Read PDO configuration for this map.
```

**Review by acolomb — CHANGES_REQUESTED** [review]
Looks good overall, just a couple of style nit-picks.

File: canopen/node/remote.py (hunk 2 above)
Comment (christiansandberg) [reply canopen/node/remote.py:158]: Agree, it is something that would be logged every time so it may not add much value.

**Review by christiansandberg — APPROVED** [review]

File: canopen/node/remote.py (hunk 2 above)
Comment (meifonglow) [reply canopen/node/remote.py:158]: Fair enough.  Removing.

Comment (meifonglow) [inline canopen/node/remote.py:158]: Removing as per consensus.
```suggestion
```

File: canopen/node/remote.py [hunk 4]
```diff
@@ -147,8 +147,15 @@ def __load_configuration_helper(self, index, subindex, name, value):
...
+        # First apply PDO configuration from object dictionary
+        self.pdo.read(from_od=True)
+        self.pdo.save()
+
+        # Now apply all other records in object dictionary
         for obj in self.object_dictionary.values():
-            if isinstance(obj, ODRecord) or isinstance(obj, ODArray):
+            if 0x1400 <= obj.index < 0x1c00:
+                # Ignore PDO related objects
+            elif isinstance(obj, ODRecord) or isinstance(obj, ODArray):
```
Comment (meifonglow) [inline canopen/node/remote.py:158]: Fixing up incorrect suggestion.

```suggestion
            if 0x1400 <= obj.index < 0x1c00:
                # Ignore PDO related objects
                pass
            elif isinstance(obj, ODRecord) or isinstance(obj, ODArray):
```

File: canopen/node/remote.py [hunk 5]
```diff
@@ -147,8 +147,16 @@ def __load_configuration_helper(self, index, subindex, name, value):
...
+        self.pdo.read(from_od=True)
+        self.pdo.save()
+
+        # Now apply all other records in object dictionary
         for obj in self.object_dictionary.values():
-            if isinstance(obj, ODRecord) or isinstance(obj, ODArray):
+            if 0x1400 <= obj.index < 0x1c00:
+                # Ignore PDO related objects
+                pass
+            elif isinstance(obj, ODRecord) or isinstance(obj, ODArray):
```
Comment (acolomb) [inline canopen/node/remote.py:159]:
```suggestion
                continue
            if isinstance(obj, ODRecord) or isinstance(obj, ODArray):
```

No need for else if you skip the iteration right away.

**Review by acolomb — APPROVED** [review]

**Pull request merged**